app/models/asset.rb

```diff
@@ -3,9 +3,11 @@
 class Asset < ApplicationRecord
   UNLOCKABLE_MODEL_PATTERN = /\ADynabook R50/
 
+  belongs_to :setting, polymorphic: true, optional: true
   validates :serial_number, :department_sold_to_id, presence: true
 
   default_scope { order(:location) }
+  scope :with_no_setting, -> { where(setting_id: nil) }
```

Comment:
I'm not 100%, but I think rails 6.1 introduced `missing`. `where.missing(:setting)` **might** work here.

Reply:
I can't find it on Rails api docs?